core/src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/slick/EntityComponent.scala

```diff
@@ -611,30 +634,32 @@ trait EntityComponent {
       }
     }
 
-    // "delete" entities by hiding and renaming
-
+    // "delete" entities by hiding and renaming. we must rename the entity to avoid future name collisions if the user
+    // attempts to create a new entity of the same name.
     def hide(workspaceContext: Workspace, entRefs: Seq[AttributeEntityReference]): ReadWriteAction[Int] = {
+      // N.B. we must hide both the entity attributes and the entity itself. Other queries, such
+      // as baseEntityAndAttributeSql, use "where ENTITY.deleted = ENTITY_ATTRIBUTE.deleted" during joins.
+      // Thus, we need to keep the "deleted" value for attributes in sync with their parent entity,
+      // when hiding that entity.
       workspaceQuery.updateLastModified(workspaceContext.workspaceIdAsUUID) andThen
-        getEntityRecords(workspaceContext.workspaceIdAsUUID, entRefs.toSet) flatMap { entRecs =>
-          val hideAction = entRecs map { rec =>
-            hideEntityAttributes(rec.id) andThen
-              hideEntityAction(rec)
-          }
-          DBIO.sequence(hideAction).map(_.sum)
-        }
+        EntityAndAttributesRawSqlQuery.batchHide(workspaceContext.workspaceIdAsUUID, entRefs) andThen
+        EntityRecordRawSqlQuery.batchHide(workspaceContext.workspaceIdAsUUID, entRefs).map( res => res.sum)
     }
 
+    // currently unused
     private def hideEntityAttributes(entityId: Long): ReadWriteAction[Seq[Int]] = {
       findActiveAttributesByEntityId(entityId).result flatMap { attrRecs =>
         DBIO.sequence(attrRecs map hideEntityAttribute)
       }
     }
 
+    // currently unused, except from hideEntityAttributes(entityId: Long) above
     private def hideEntityAttribute(attrRec: EntityAttributeRecord): WriteAction[Int] = {
       val currentTime = new Timestamp(new Date().getTime)
       entityAttributeQuery.filter(_.id === attrRec.id).map(rec => (rec.deleted, rec.name, rec.deletedDate)).update(true, renameForHiding(attrRec.id, attrRec.name), Option(currentTime))
     }
 
+    // currently unused
```

Comment:
should I just delete these three unused methods?

Reply:
also in favor of removing

Reply:
removed!!